packages/mana/base.go

```diff
@@ -34,17 +34,22 @@ func (bm *BaseMana) updateEBM1(n time.Duration) {
 }
 
 func (bm *BaseMana) updateBM2(n time.Duration) {
-	bm.BaseMana2 = bm.BaseMana2 * math.Pow(math.E, -decay*n.Seconds())
+	bm.BaseMana2 = bm.BaseMana2 * math.Pow(math.E, -Decay*n.Seconds())
+}
+
+// GetBM2 computes and returns the EBM2 based on the value and duration specified.
+func GetBM2(value float64, n time.Duration) float64 {
+	return value * math.Pow(math.E, -Decay*n.Seconds())
```

Comment:
this doesn't make sense: `GetBM2` that returns `EBM2`?
We want to calculate the `pending mana`, that is, the amount of `BM2` that would be pledged at time `t` by spending the output. It has nothing to do with `EBM2`, and is calculated differently.
`value * math.Pow(math.E, -Decay*n.Seconds())` gives you a decreasing function, but we need the opposite. The correct equation is on line 96.
Rename this function to `GetPendingMana` or smthing more descriptive, with the calculation from line 96. Or, since now you have `Decay` exposed, you could place this calculation anywhere. Functions that are not used in `base.go` (and don't need data from there) should be in the plugin imho.